horton/meanfield/builtin.py

```diff
@@ -112,47 +125,47 @@ def __init__(self, label='x_dirac', coeff=None):
         GridObservable.__init__(self, label)
 
     def _update_pot(self, cache, grid, select):
-        '''Recompute an Exchange potential if invalid
-
-           cache
-                A cache instance in which the potential is stored.
-
-           grid
-                The integration grid.
-
-           select
-                'alpha' or 'beta'
-        '''
-        rho = cache['rho_%s' % select]
+        """Recompute an Exchange potential if invalid.
+
+        Parameters
+        ----------
+        cache : Cache
+            Storage for the potentials.
+        grid : IntGrid
+            A numerical integration grid.
+        select : str
+            'alpha' or 'beta'
+        """
+        rho = cache['all_%s' % select][:, 0]
```

Comment:
The rather generic name `all` is used because it is an array that may contain different types of density-related data:
* LDA: only the density
* GGA: density, deriv_x, deriv_y, deriv_z
* MGGA: density, deriv_x, deriv_y, deriv_z, laplacian, kinetic energy density